include/modem/nrf_modem_lib_trace.h

```diff
@@ -51,6 +76,42 @@ int nrf_modem_lib_trace_processing_done_wait(k_timeout_t timeout);
  */
 int nrf_modem_lib_trace_level_set(enum nrf_modem_lib_trace_level trace_level);
 
+/**
+ * @brief Get the number of bytes stored in the compile-time selected trace backend.
+ *
+ * @note To ensure the retrieved number is correct, this function should only be called
+ *       when the backend is done processing traces.
+ *
+ * @returns Number of bytes stored in the trace backend.
+ */
+size_t nrf_modem_lib_trace_data_size(void);
+
+/**
+ * @brief Read trace data
+ *
+ * Read out the trace data. After read, the backend can choose to free the
+ * space and prepare it for further write operations.
+ *
+ * @note This operation is only supported with some trace backends.
+ *
+ * @param buf Output buffer
+ * @param len Size of output buffer
+ *
+ * @return read number of bytes, negative errno on failure.
+ */
+int nrf_modem_lib_trace_read(uint8_t *buf, size_t len);
+
+/**
+ * @brief Clear captured trace data
+ *
+ * Clear all captured trace data and prepare for capturing new traces.
+ *
+ * @note This operation is only supported with some trace backends.
```

Comment:
```suggestion
 * @note This operation is implemented only by some trace backends.
```

Same for similar notes.

However, I'd argue we shouldn't add this note to the optional functions. This is the public API and an implementer doesn't really care what other implementations are or aren't implementing.

Reply:
I think it is important for the application engineer to know that there are different backends with different behaviors. There might be cases where someone might want to change the backend during development.

Reply:
I think supported is ok to use here. If the backend does not support the feature, `nrf_modem_lib_trace.c` returns with `-ENOTSUP`.

Reply:
It would be nice to update the doc-comment that it returns `-ENOTSUP` if it's not supported by the backend. Same for the other similar functions.

Reply:
updated